src/plots/plots.js

```diff
@@ -540,6 +544,23 @@ plots.supplyDefaults = function(gd) {
             (gd.calcdata[i][0] || {}).trace = trace;
         }
     }
+
+    // Set up the default keyframe if it doesn't exist:
+    if(!gd._frameData) {
+        gd._frameData = {};
+    }
+
+    if(!gd._frameData._frames) {
+        gd._frameData._frames = [];
+    }
+
+    if(!gd._frameData._frameHash) {
+        gd._frameData._frameHash = {};
+    }
+
+    if(!gd._frameData._counter) {
+        gd._frameData._counter = 0;
+    }
```

Comment:
Yeah, the first step in *every* frame method was the nested if-not-defined-then-initialize step, so seemed cleaner to just initialize and take it for granted that it's present.

Hmm… I guess there must be a mechanism that stops plotly from loading data from JSON on every `.plot` call, so I should probably use the same mechanism to unpack the frames on first initialization.